Integrated yarn and upgraded to python 3

fahdi opened the pull request:
1. Integrated yarn, so third party libs are installed automatically. 
2. Updated docker to improve the image building (FROM python:onbuild)
3. Integrated yarn into docker 
4. General cleanup

fahdi committed:
Removing CSS files. They are going to be added via package management.

fahdi committed:
.gitignore added

fahdi committed:
Redundant CSS file deleted. Cleaned up the main CSS file.

fahdi committed:
Updating gitignore

fahdi committed:
Making the app python 3 compatible

fahdi committed:
Changes
1. Added yarn for package management.
2. Using node modules in the layout file i.e. layout.html
3. Updated docker with a better python image that uses python 3.6 and takes requirements file for automatic installs.
4. Updated the main css file to import bootstrap and font awesome into it directly.

fahdi committed:
Added the spectrum as a yarn dependency and updated the respective template code.

fahdi committed:
Masonry layout and jQuery added and yarn dependencies. Template files modified accordingly.

fahdi committed:
Cleaning up Dockerfile

fahdi committed:
Adding a docker ignore file

fahdi committed:
Fixing the order of script loading

ahmednooor commented on templates/layout.html:
```diff
@@ -6,8 +6,6 @@
     <meta name="viewport" content="width=device-width, initial-scale=1.0">
     <meta http-equiv="X-UA-Compatible" content="ie=edge">
     <title>{% block title %}{% endblock %}</title>
-    <link rel="stylesheet" type="text/css" href="../static/css/bootstrap.min.css">
```
How the css files of bootstrap and fontawesome will be linked?

fahdi replied:
On docker, everything is automated by Dockerfile. 

If you want to run the project manually using system installed stuff, that will be taken care of by the [shell script](https://github.com/fahdi/palettro/blob/c6ba5022ec31f7891438cd2f6ecfe355c850be10/build.sh).  You can see the commands and run them manually if you like.

fahdi replied:
If you check in main.css, it has the respective import. I'd later on automate it all, so they are all combined and minified for faster delivery. Just thinking of a best way to package the whole app.

ahmednooor replied:
leave the spectrum css file as it is (or reincorporate the old spectrum.css file in next commit) coz i modified it a bit. so in case of downloading the latest file might not have the same modifications etc.

fahdi replied:
I plan a SASS integration which will make one compressed CSS file. The spectrum changes can be a part of it.

ahmednooor replied:
ok lets see.

fahdi committed:
Adding an install shell script which should make it easy for setting up. Updated the README accordingly

fahdi commented:
I have added a shell script install.sh and updated the README accordingly. No matter if you use docker or not, it will take care of the setup for you. The readme also has manual setup instructions.

fahdi committed:
Updating install.sh to be more concrete
